File: frontend/src/modules/proyectos/application/ActualizarAsignacionRolUseCase.ts
import { ProyectoMiembroRepositoryPort } from '../domain/ProyectoMiembroRepositoryPort';
import { ProyectoMiembro } from '../domain/ProyectoMiembro';

export class ActualizarAsignacionRolUseCase {
  constructor(private repo: ProyectoMiembroRepositoryPort) {}
  async ejecutar(
    proyectoId: string,
    asignacionId: string,
    rolId: string,
    componenteId?: string | null,
    accionId?: string | null
  ): Promise<ProyectoMiembro> {
    return this.repo.actualizarAsignacionRol(proyectoId, asignacionId, rolId, componenteId, accionId);
  }
}

File: backend/modulos/miembros/aplicacion/ActualizarAsignacionRolUseCase.py
from modulos.miembros.dominio.AsignacionMiembroRolRepositoryPort import AsignacionMiembroRolRepositoryPort

class ActualizarAsignacionRolUseCase:
    def __init__(self, repo: AsignacionMiembroRolRepositoryPort):
        self.repo = repo

    def ejecutar(
        self,
        proyecto_id: str,
        asignacion_id: str,
        rol_id: str,
        componente_id: str = None,
        accion_id: str = None
    ):
        return self.repo.actualizar_asignacion_rol(
            proyecto_id=proyecto_id,
            asignacion_id=asignacion_id,
            rol_id=rol_id,
            componente_id=componente_id,
            accion_id=accion_id
        )
